src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -570,22 +845,22 @@ function AnalyticsService:LogMarketplacePurchase(data: MarketplacePurchaseEvent)
 		elseif not data.cartType then
 			return reject("cartType is required")
 		end
-		
-		GameAnalytics:addBusinessEvent(data.userId, {
+
+		self:_wrapper("addBusinessEvent", data.userId, {
 			itemType = data.itemType,
-			itemId = data.id,
-			amount = data.amount or 1,
-			cartType = data.cartType
+			itemId = typeof(data.id) == "number" and tostring(data.id) or data.id,
+			amount = data.robuxPrice or 0,
+			cartType = data.cartType,
 		})
-		
-		return resolve()
+
+		resolve()
```

Comment:
```suggestion
		return resolve()
```

The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.